usort/translate.py

```diff
@@ -45,48 +183,216 @@ def import_from_node(node: cst.SimpleStatementLine, config: Config) -> SortableI
+) -> cst.BaseStatement:
+    width = 88  # TODO: get width from tool.black
+    node = import_to_node_single(imp, module)
+    content = render_node(node, module)
+    if len(content) > width:
+        node = import_to_node_multi(imp, module)
+    return node
+
+
+def import_to_node_single(imp: SortableImport, module: cst.Module) -> cst.BaseStatement:
+    leading_lines = [
+        cst.EmptyLine(
+            indent=True, comment=(cst.Comment(line) if line.startswith("#") else None)
+        )
+        for line in imp.comments.before
+    ]
+
+    trailing_whitespace = cst.TrailingWhitespace()
+    trailing_comments = list(imp.comments.first_inline)
+
+    names: List[cst.ImportAlias] = []
+    for item in imp.items:
+        name = name_to_node(item.name)
+        asname = cst.AsName(name=cst.Name(item.asname)) if item.asname else None
+        node = cst.ImportAlias(name=name, asname=asname)
+        names.append(node)
+        trailing_comments += item.comments.before
+        trailing_comments += item.comments.inline
+        trailing_comments += item.comments.following
+
+    trailing_comments += imp.comments.last_inline
+    if trailing_comments:
+        text = COMMENT_INDENT.join(trailing_comments)
+        trailing_whitespace = cst.TrailingWhitespace(
+            whitespace=cst.SimpleWhitespace(COMMENT_INDENT), comment=cst.Comment(text)
+        )
+
+    if imp.stem:
+        stem, ndots = split_relative(imp.stem)
+        if not stem:
+            module_name = None
+        else:
+            module_name = name_to_node(stem)
+        relative = (cst.Dot(),) * ndots
+
+        line = cst.SimpleStatementLine(
+            body=[cst.ImportFrom(module=module_name, names=names, relative=relative)],
+            leading_lines=leading_lines,
+            trailing_whitespace=trailing_whitespace,
+        )
+
+    else:
+        line = cst.SimpleStatementLine(
+            body=[cst.Import(names=names)],
+            leading_lines=leading_lines,
+            trailing_whitespace=trailing_whitespace,
+        )
+
+    return line
+
+
+def import_to_node_multi(imp: SortableImport, module: cst.Module) -> cst.BaseStatement:
+    body: List[cst.BaseSmallStatement] = []
+    names: List[cst.ImportAlias] = []
+    prev: Optional[cst.ImportAlias] = None
+    lpar_lines: List[cst.EmptyLine] = []
+    lpar_inline: cst.TrailingWhitespace = cst.TrailingWhitespace()
+
+    for idx, item in enumerate(imp.items):
+        name = name_to_node(item.name)
+        asname = cst.AsName(name=cst.Name(item.asname)) if item.asname else None
+
+        # Leading comments actually have to be trailing comments on the previous node.
+        # That means putting them on the lpar node for the first item
+        if item.comments.before:
+            lines = [
+                cst.EmptyLine(
+                    indent=True,
+                    comment=cst.Comment(c),
+                    whitespace=cst.SimpleWhitespace(BLOCK_INDENT),
```

Comment:
Per https://libcst.readthedocs.io/en/latest/nodes.html#libcst.Module mod.default_indent should be inferred from the file (and I assume defaults to four spaces).  This and correct newlines are actually why I plumbed through module all the way here.